__init__.py

```diff
@@ -68,41 +69,69 @@ def _download_and_extract(split):
-        _download_and_extract(split)
-    else:
-        for _split in _SPLIT_TO_FILE_NAME:
-            _download_and_extract(_split)
-
-    dataset_type = fo.types.COCODetectionDataset
-    num_samples = 0
-    classes = 0
-    if split:
-        split_data_dir = os.path.join(dataset_dir, split, "data")
-        num_samples += len(etau.list_files(split_data_dir, recursive=True))
-        classes = max(
-            classes, len(etau.list_subdirs(split_data_dir, recursive=True)) - 1
-        )
-    else:
-        for d in etau.list_subdirs(dataset_dir):
-            split_data_dir = os.path.join(dataset_dir, d, "data")
-            if os.path.exists(split_data_dir):
-                num_samples += len(
-                    etau.list_files(split_data_dir, recursive=True)
-                )
-            classes = max(
-                classes,
-                len(etau.list_subdirs(split_data_dir, recursive=True)) - 1,
-            )
+    if split not in _SPLIT_TO_FILE_NAME.keys():
+        raise IOError(f"Input split {split} is not supported.")
+    _download_and_extract(split)
+
+    dataset_type = None
+    split_data_dir = os.path.join(dataset_dir, split, "data")
+    num_samples = len(etau.list_files(split_data_dir, recursive=True))
+    classes = len(etau.list_subdirs(split_data_dir, recursive=True)) - 1
 
     shutil.rmtree(scratch_dir, ignore_errors=True)
     shutil.rmtree(os.path.join(dataset_dir, "__pycache__"), ignore_errors=True)
 
     return dataset_type, num_samples, classes
 
 
+def load_dataset(dataset, dataset_dir, split):
+    """Loads the dataset into the given FiftyOne dataset.
+
+    Args:
+        dataset: a :class:`fiftyone.core.dataset.Dataset` to which to import
+        dataset_dir: the directory to which the dataset was downloaded
+        split: a split to load. The supported values are
+            ``("train", "train-mini", "validation", "test")``
+    """
+    split_dir = os.path.join(dataset_dir, split)
+    dataset.add_dir(
+        dataset_dir=split_dir,
+        dataset_type=fo.types.COCODetectionDataset,
+    )
+
+    labels_json = os.path.join(split_dir, "labels.json")
+    labels = etas.load_json(labels_json)
+    _add_classification_labels(
+        dataset, labels=labels["annotations"], categories=labels["categories"]
+    )
+
+
+def _add_classification_labels(dataset, labels, categories):
+    coco_objects = []
+    for d in labels:
+        # Add pseudo bbox for detection to classification conversion
+        d["bbox"] = [0, 0, 0, 0]
+        coco_obj = COCOObject.from_anno_dict(d)
+        coco_objects.append([coco_obj])
+
+    dataset.compute_metadata()
+    widths, heights = dataset.values(["metadata.width", "metadata.height"])
+    classes_map = {c["id"]: c["name"] for c in categories}
+
+    classification_labels = []
+    for _coco_objects, width, height in zip(coco_objects, widths, heights):
+        frame_size = (width, height)
+        _detections = _coco_objects_to_detections(
```

Comment:
Using a private method of an FO utility module in a third-party repo is dangerous. Core FO developers may rename it at any time.